### ISSUE TITLE
Build fails with: use of internal package not allowed

### ISSUE THREAD

**Issue by mikkeloscar** [prasmussen/gdrive#81]
The build fails with the following error with go 1.5:

```
package github.com/prasmussen/gdrive/cli
	imports github.com/prasmussen/gdrive/gdrive
	imports github.com/prasmussen/gdrive/auth
	imports github.com/prasmussen/gdrive/util
	imports github.com/prasmussen/google-api-go-client/drive/v2
	imports github.com/prasmussen/google-api-go-client/googleapi
	imports google.golang.org/api/googleapi/internal/uritemplates: use of internal package not allowed
```

This is due to the reference to the upstream repo here: https://github.com/prasmussen/google-api-go-client/blob/master/googleapi/googleapi.go#L27

Changing the import to:

```
github.com/prasmussen/google-api-go-client/googleapi/internal/uritemplates
```

solves the issue.

**Comment by Drezzler** [comment]
This an old issue , but the solution is to use golang v1.4.2 , as the versions past that have a bug .

**Comment by mikkeloscar** [comment]
Any chance you could point me to a discussion of this bug in 1.5? I can only find a somewhat related bug golang/go#11307 but it has been fixed and is only related to importing local internal packages, which does work in 1.5.

**Comment by Drezzler** [comment]
I am sorry , but i had not read your first post carefully , but indeed that reference to google api is broken .

Is their any particular advantage of using go v1.5 instead of 1.4.2 for gdrive build ?

**Comment by mikkeloscar** [comment]
I don't know if there is any advantage, but since v1.5 is the current stable release of golang I think it makes sense to keep gdrive (or anything else) up to date.

I don't know if my suggested solution is the right way to fix it, that's why I haven't made a PR.

**Comment by kevin-cantwell** [comment]
You might want to give 1.5.2 a test drive. It appears to not have this bug.

**Comment by mikkeloscar** [comment]
I see the same issue:

```
$ go get
package .
	imports github.com/prasmussen/gdrive/cli
	imports github.com/prasmussen/gdrive/gdrive
	imports github.com/prasmussen/gdrive/auth
	imports github.com/prasmussen/gdrive/util
	imports github.com/prasmussen/google-api-go-client/drive/v2
	imports github.com/prasmussen/google-api-go-client/googleapi
	imports google.golang.org/api/googleapi/internal/uritemplates: use of internal package not allowed
$ go version
go version go1.5.2 linux/amd64
```

**Comment by ksimka** [comment]
Exactly the same issue with 1.5.2

**Comment by labkode** [comment]
@Drezzler  for me the big advantage of 1.5* is the vendor experiment support

**Comment by avinash-** [comment]
Same issue with go version 1.5.3

package github.com/prasmussen/gdrive/cli
	imports github.com/prasmussen/gdrive/gdrive
	imports github.com/prasmussen/gdrive/auth
	imports github.com/prasmussen/gdrive/util
	imports github.com/prasmussen/google-api-go-client/drive/v2
	imports github.com/prasmussen/google-api-go-client/googleapi
	imports google.golang.org/api/googleapi/internal/uritemplates: use of internal package not allowed
[ventanni@oc0283078624 gdrive]$ go version
go version go1.5.3 linux/amd64

@mikkeloscar Can you help me in solving this.

**Comment by mikkeloscar** [comment]
@avinash- have you tried to change the line as described in my original post? That should still fix the problem.

**Comment by avinash-** [comment]
@mikkeloscar I can't change the main file because I need write access. 
Plz guide me how to change in which file I need to change ?

Thanks for Replying @mikkeloscar :)

[ventanni@oc0283078624 gdrive]$ go build drive.go 
drive.go:5:2: cannot find package "github.com/prasmussen/gdrive/cli" in any of:
	/usr/local/go/src/github.com/prasmussen/gdrive/cli (from $GOROOT)
	/home/ventanni/Avinash/gdrive/src/github.com/prasmussen/gdrive/cli (from $GOPATH)
drive.go:6:2: cannot find package "github.com/prasmussen/gdrive/gdrive" in any of:
	/usr/local/go/src/github.com/prasmussen/gdrive/gdrive (from $GOROOT)
	/home/ventanni/Avinash/gdrive/src/github.com/prasmussen/gdrive/gdrive (from $GOPATH)
drive.go:7:2: cannot find package "github.com/prasmussen/gdrive/util" in any of:
	/usr/local/go/src/github.com/prasmussen/gdrive/util (from $GOROOT)
	/home/ventanni/Avinash/gdrive/src/github.com/prasmussen/gdrive/util (from $GOPATH)
drive.go:8:2: cannot find package "github.com/prasmussen/google-api-go-client/googleapi" in any of:
	/usr/local/go/src/github.com/prasmussen/google-api-go-client/googleapi (from $GOROOT)
	/home/ventanni/Avinash/gdrive/src/github.com/prasmussen/google-api-go-client/googleapi (from $GOPATH)
drive.go:9:2: cannot find package "github.com/prasmussen/google-api-go-client/googleapi/internal/uritemplates" in any of:
	/usr/local/go/src/github.com/prasmussen/google-api-go-client/googleapi/internal/uritemplates (from $GOROOT)
	/home/ventanni/Avinash/gdrive/src/github.com/prasmussen/google-api-go-client/googleapi/internal/uritemplates (from $GOPATH)
drive.go:10:2: cannot find package "github.com/voxelbrain/goptions" in any of:
	/usr/local/go/src/github.com/voxelbrain/goptions (from $GOROOT)
	/home/ventanni/Avinash/gdrive/src/github.com/voxelbrain/goptions (from $GOPATH)
[ventanni@oc0283078624 gdrive]$ ld
ld: no input files
[ventanni@oc0283078624 gdrive]$ ls
auth  cli  config  drive.go  gdrive  LICENSE  README.md  _release  util
[ventanni@oc0283078624 gdrive]$ go get github.com/prasmussen/gdrive
warning: code.google.com is shutting down; import path code.google.com/p/goauth2/oauth will stop working
package github.com/prasmussen/gdrive
	imports github.com/prasmussen/gdrive/cli
	imports github.com/prasmussen/gdrive/gdrive
	imports github.com/prasmussen/gdrive/auth
	imports github.com/prasmussen/gdrive/util
	imports github.com/prasmussen/google-api-go-client/drive/v2
	imports github.com/prasmussen/google-api-go-client/googleapi
	imports google.golang.org/api/googleapi/internal/uritemplates: use of internal package not allowed

**Comment by mikkeloscar** [comment]
@avinash- 
It looks like you need to change this file: `/home/ventanni/Avinash/gdrive/src/github.com/prasmussen/google-api-go-client/googleapi/googleapi.go` assuming `/home/ventanni/Avinash/gdrive` is your `$GOPATH`.

This is the changes you need to make:

```diff
diff --git a/googleapi/googleapi.go b/googleapi/googleapi.go
index c4d0b93..58077a0 100644
--- a/googleapi/googleapi.go
+++ b/googleapi/googleapi.go
@@ -23,8 +23,8 @@ import (
 	"sync"
 	"time"
 
+	"github.com/prasmussen/google-api-go-client/googleapi/internal/uritemplates"
 	"golang.org/x/net/context"
-	"google.golang.org/api/googleapi/internal/uritemplates"
 )
 
 // ContentTyper is an interface for Readers which know (or would like
```

**Comment by avinash-** [comment]
@mikkeloscar Thanks Yor are Super you made my day. :+1: 

[ventanni@oc0283078624 gdrive]$ go build drive.go 
[ventanni@oc0283078624 gdrive]$ drive list 
Command not found. 

I think I need  to replace 'drive' binary generated and place it in '/usr/local/bin' , make the binary executable with 'chmod +x drive'

I think they are in "/home/ventanni/Avinash/gdrive/src"

**Comment by avinash-** [comment]
@mikkeloscar Help me in configuring Binaries 
 
Where are the compiled binary located to place in "/usr/local/bin"

[ventanni@oc0283078624 gdrive]$ ls
auth  cli  config  drive  drive.go  gdrive  LICENSE  pkg  README.md  _release  src  util
[ventanni@oc0283078624 gdrive]$ go build drive.go 
[ventanni@oc0283078624 gdrive]$ ls
auth  cli  config  drive  drive.go  gdrive  LICENSE  pkg  README.md  _release  src  util
[ventanni@oc0283078624 gdrive]$ drive list
Command not found.

**Comment by mikkeloscar** [comment]
@avinash- the binary is right there. It's the one called `drive`. You can run it with `./drive` or copy it like you suggest.

**Comment by avinash-** [comment]
@mikkeloscar Thanks Again it Worked 
How han we reset API, I need to switch google drive account with new gmail ID ?

[ventanni@oc0283078624 gdrive]$ drive list
googleapi: Error 401: Invalid Credentials, authError

**Comment by willemw12** [comment]
Works now (at least with go 1.6). Close this issue?

**Comment by mikkeloscar** [comment]
It was fixed here: prasmussen/google-api-go-client#1
